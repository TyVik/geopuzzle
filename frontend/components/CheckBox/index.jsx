'use strict';
import React from 'react';


class CheckBox extends React.Component {
    render() {
        return <input id={this.props.id} value={this.props.id} type="checkbox" name={this.props.name}
                      onChange={this.props.onChange} checked={this.props.checked} />;
    }
}

export default CheckBox;
