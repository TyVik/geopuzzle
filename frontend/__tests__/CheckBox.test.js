import React from 'react';
import {shallow} from 'enzyme';
import CheckBox from '../components/CheckBox';


describe('shallow <CheckBox /> components', () => {
  let onChange = jest.fn();
  let props = {id: 0, name: 'test', checked: false, onChange: onChange};
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CheckBox {...props}/>);
  });

  it('render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('check state', () => {
    let attrs = wrapper.props();
    expect(attrs).toHaveProperty('name', 'test');
    expect(attrs).toHaveProperty('checked', false);
    expect(attrs).toHaveProperty('value', 0);
  });

  it('click', () => {
    wrapper.simulate('change');
    expect(onChange).toHaveBeenCalled();
    expect(wrapper.props()).toHaveProperty('checked', false);  // checked should be required by props
  });
});
