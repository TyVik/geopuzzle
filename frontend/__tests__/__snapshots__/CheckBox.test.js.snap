// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow <CheckBox /> components render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CheckBox
    checked={false}
    id={0}
    name="test"
    onChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "checked": false,
      "id": 0,
      "name": "test",
      "onChange": [MockFunction],
      "type": "checkbox",
      "value": 0,
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "checked": false,
        "id": 0,
        "name": "test",
        "onChange": [MockFunction],
        "type": "checkbox",
        "value": 0,
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
